control-plane/subcommand/gateway-resources/command_test.go

```diff
@@ -254,3 +268,95 @@ func TestRun(t *testing.T) {
 		})
 	}
 }
+
+var meshGWConfigFileContent = `gatewayClassConfigs:
+  - apiVersion: mesh.consul.hashicorp.com/v2beta1
+    metadata:
+      name: consul-mesh-gateway
+      namespace: namespace
+    kind: gatewayClassConfig
+    spec:
+      deployment:
+meshGateways:
+  - name: mesh-gateway
+    spec:
+      gatewayClassName: consul-mesh-gateway
+`
+
+func TestRun_loadMeshGatewayConfigs(t *testing.T) {
+	filename := createMeshGatewayConfigFile(t)
+	// setup k8s client
+	s := runtime.NewScheme()
+	require.NoError(t, gwv1beta1.Install(s))
+	require.NoError(t, v1alpha1.AddToScheme(s))
+
+	client := fake.NewClientBuilder().WithScheme(s).Build()
+
+	ui := cli.NewMockUi()
+	cmd := Command{
+		UI:                            ui,
+		k8sClient:                     client,
+		flagMeshGatewayConfigLocation: filename,
+	}
+
+	err := cmd.loadMeshGatewayConfigs()
+	require.NoError(t, err)
+	require.NotEmpty(t, cmd.meshGatewayConfig.GatewayClassConfigs)
+
+	// we only created one class config
+	classConfig := cmd.meshGatewayConfig.GatewayClassConfigs[0].DeepCopy()
+
+	// TODO: Add resources to the example yaml and test here once https://github.com/hashicorp/consul/pull/19725 merges
+	expectedClassConfig := v2beta1.GatewayClassConfig{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "mesh.consul.hashicorp.com/v2beta1",
+			Kind:       "gatewayClassConfig",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "consul-mesh-gateway",
+			Namespace: "namespace",
+		},
+		Spec:   meshv2beta1.GatewayClassConfig{},
+		Status: v2beta1.Status{},
+	}
+	require.Equal(t, expectedClassConfig.DeepCopy(), classConfig)
+}
+
+func TestRun_loadMeshGatewayConfigsWhenConfigFileDoesNotExist(t *testing.T) {
+	filename := "./consul/config/config.yaml"
+	s := runtime.NewScheme()
+	require.NoError(t, gwv1beta1.Install(s))
+	require.NoError(t, v1alpha1.AddToScheme(s))
+
+	client := fake.NewClientBuilder().WithScheme(s).Build()
+
+	ui := cli.NewMockUi()
+	cmd := Command{
+		UI:                            ui,
+		k8sClient:                     client,
+		flagMeshGatewayConfigLocation: filename,
+	}
+
+	err := cmd.loadMeshGatewayConfigs()
+	require.NoError(t, err)
```

Comment:
we should probably verify this warning is triggered https://github.com/hashicorp/consul-k8s/pull/3273/files#diff-5234182194c12306a3b7a58b33e6908998d552b2ba30d1f42cde158645864c55R367

Reply:
done and done